enhancements/aiops/20230120-Identify-prometheus-metrics-that-match-known-anomaly-chart-patterns.md

```diff
@@ -0,0 +1,57 @@
+# Title: 
+Identify prometheus metrics that match known anomaly chart patterns
+
+## Summary: 
+This enhancement proposal offers a new feature to match prometheus metrics to a set of known pre-defined anomaly chart patterns. This feature will make operators' life easier in the process of outage diagnosis.
+
+## Use case: 
+* Monitoring metrics in a multi-cluster setup is challenging. Operators could be responsible for hundreds of clusters and each cluster has tons of different metrics. When an outage occurs, it's difficult to identify root-cause metrics to assist outage diagnosis. Users need tools to help them reduce the scope and identify anomaly metrics faster.
+
+## Benefits: 
+* Makes operators' life easier in the process of outage diagnosis.
+* Improve user experience. Metric data will get condensed and categorized into insights that are much easier to digest.
+* Reduce mean time to resolve outages. This method will shorten the time operators taken to find out root cause.
+* Reduce economic loss.
+
+## Impact: 
+* No direct impact to existing system in terms of performance. However, initially this feature will only be deployed in `reactive mode` -- it's only triggered when an alert is fired. If this new feature is deployed in a `proactive mode`(which means it runs periodically) in the future, then it might require more resources as it would have to continuously compute for all metrics.
+
+## Implementation details: 
+* Pattern classification process. Inspired by this paper, https://netman.aiops.org/wp-content/uploads/2021/10/wch_ISSRE-1.pdf. 
+	a. A simple metric anomaly detection algorithm, that predicts every metric either normal or anomaly. An example method is Kolmogorov-Smirnov test.
+	b. A machine-learning or deep-learning model will be pre-trained for pattern classification. This would require some real abnormal metric data with human labels, and then apply data augmentation techniques to them. Once model is trained, it will be applied to anomaly metrics from step a and categorize them into a few pre-defined patterns.
+	c. Analyse report. A report should be generated to summarize the notable patterns.
+* Metric data collection. Data will be pulled from Cortex as chunks. Likely it would require data from last a few hours before an outage.
+* Back-end apis for frontend to connect to.
+* UI. 
+	a. User would need to manually trigger this feature, so there needs to buttons attached to each alerts to trigger this analysis.
+	b. The analyse report should be visualized in UI.
+
+## Acceptance criteria: 
+* Pattern classification accuracy should be good. Threshold to be decided. 
```

Comment:
"good" is not descriptive enough. I need a threshold before approving

Reply:
the reason I don't want to set an arbitrarily threshold is we don't even have any real data yet, plus it would depend on the quality of data . it you really need one I'd propose `98%` which is the testing accuracy got on the egg Kaggle dataset I found online.